### ISSUE TITLE
BUG: Strange error while describing the TreeExplainer for catboost

### ISSUE THREAD

**Issue by rakshitrao99** [shap/shap#3176]
### Issue Description

I'm running a ml model, but I keep getting this error: 

> AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:

I have checked the version of shap and numpy as well: The version are as follows: 
('0.41.0', '1.24.3')

### Minimal Reproducible Example

```python
import shap
num_cols is a list of numerical columns and cat_cols is a list of categorical columns
catb is a catboost model object:
%%time
catb = CatBoostClassifier(iterations=100, random_seed=42, learning_rate=0.5)
catb.fit(df_train_data[num_cols+cat_cols], df_train_data[['liability_ind']] , cat_features=cat_cols, verbose=True)

explainerModel_prob = shap.TreeExplainer(catb, 
                                         data = shap.sample(df_train_data[num_cols+cat_cols], 1000), 
                                         model_output='probability')
```

### Expected Behavior

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[148], line 1
----> 1 explainerModel_prob = shap.TreeExplainer(catb, data = shap.sample(df_train_data[num_cols+cat_cols], 1000), model_output='probability')

File ~\.conda\envs\myenv\Lib\site-packages\shap\explainers\_tree.py:101, in Tree.__init__(self, model, data, model_output, feature_perturbation, feature_names, approximate, **deprecated_options)
     98     self.data_feature_names = list(data.columns)
    100 masker = data
--> 101 super(Tree, self).__init__(model, masker, feature_names=feature_names)
    103 if type(self.masker) is maskers.Independent:
    104     data = self.masker.data

File ~\.conda\envs\myenv\Lib\site-packages\shap\explainers\_explainer.py:88, in Explainer.__init__(self, model, masker, link, algorithm, output_names, feature_names, linearize_link, seed, **kwargs)
     86         self.masker = maskers.Partition(masker)
     87     else:
---> 88         self.masker = maskers.Independent(masker)
     89 elif safe_isinstance(masker, ["transformers.PreTrainedTokenizer", "transformers.tokenization_utils_base.PreTrainedTokenizerBase"]):
     90     if is_transformers_lm(self.model):
     91         # auto assign text infilling if model is a transformer model with lm head

File ~\.conda\envs\myenv\Lib\site-packages\shap\maskers\_tabular.py:262, in Independent.__init__(self, data, max_samples)
    247 def __init__(self, data, max_samples=100):
    248     """ Build a Independent masker with the given background data.
    249 
    250     Parameters
   (...)
    260         about 1, 10, 100, or 1000 background samples are reasonable choices.
    261     """
--> 262     super().__init__(data, max_samples=max_samples, clustering=None)

File ~\.conda\envs\myenv\Lib\site-packages\shap\maskers\_tabular.py:82, in Tabular.__init__(self, data, max_samples, clustering)
     80 # self._last_mask = np.zeros(self.data.shape[1], dtype=np.bool)
     81 self._masked_data = data.copy()
---> 82 self._last_mask = np.zeros(data.shape[1], dtype=np.bool)
     83 self.shape = self.data.shape
     84 self.supports_delta_masking = True

File ~\.conda\envs\myenv\Lib\site-packages\numpy\__init__.py:305, in __getattr__(attr)
    300     warnings.warn(
    301         f"In the future `np.{attr}` will be defined as the "
    302         "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    304 if attr in __former_attrs__:
--> 305     raise AttributeError(__former_attrs__[attr])
    307 # Importing Tester requires importing all of UnitTest which is not a
    308 # cheap import Since it is mainly used in test suits, we lazy import it
    309 # here to save on the order of 10 ms of import time for most users
    310 #
    311 # The previous way Tester was imported also had a side effect of adding
    312 # the full `numpy.testing` namespace
    313 if attr == 'testing':

AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.41.0

**Comment by ivanightingale** [comment]
I suggest renaming the issue to "Update `np.bool` to `bool`".

EDIT: the latest stable version already have `np.bool` updated to `bool`. I suggest closing the issue.

**Comment by znacer** [comment]
I did not manage to reproduce this issue.
Could you give a complete minimal reproducibility example ?
I tried with the following : 
```py
import numpy as np
import pandas as pd
import shap
from catboost import CatBoostClassifier

# train a simple catboost model
df_train_data,y = shap.datasets.adult()
cat_cols = df_train_data.select_dtypes(include=["int8", "int64"]).columns.tolist()
num_cols = df_train_data.select_dtypes(include=["float64"]).columns.tolist()

%%time
catb = CatBoostClassifier(iterations=100, random_seed=42, learning_rate=0.5)
catb.fit(df_train_data[num_cols+cat_cols], y , cat_features=cat_cols, verbose=True)

explainerModel_prob = shap.TreeExplainer(catb, 
                                         data = shap.sample(df_train_data[num_cols+cat_cols], 1000), 
                                         model_output='probability')
```

**Comment by ArbyC** [comment]
I am getting the same error with 

```
masker = shap.maskers.Independent(data = train_x, max_samples=1000)
explainer = shap.Explainer(ebm, masker = masker)
shap_values = explainer(test_x)
```
numpy version 1.24.3
Shap version 0.41.0

The code ran smoothly before. 

Error:

```
E:\Anaconda\lib\site-packages\shap\maskers\_tabular.py:82: FutureWarning:

In the future `np.bool` will be defined as the corresponding NumPy scalar.

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[35], line 1
----> 1 masker = shap.maskers.Independent(data = train_x, max_samples=1000)
      2 explainer = shap.Explainer(ebm, masker = masker)
      3 shap_values = explainer(test_x)

File E:\Anaconda\lib\site-packages\shap\maskers\_tabular.py:262, in Independent.__init__(self, data, max_samples)
    247 def __init__(self, data, max_samples=100):
    248     """ Build a Independent masker with the given background data.
    249 
    250     Parameters
   (...)
    260         about 1, 10, 100, or 1000 background samples are reasonable choices.
    261     """
--> 262     super().__init__(data, max_samples=max_samples, clustering=None)

File E:\Anaconda\lib\site-packages\shap\maskers\_tabular.py:82, in Tabular.__init__(self, data, max_samples, clustering)
     80 # self._last_mask = np.zeros(self.data.shape[1], dtype=np.bool)
     81 self._masked_data = data.copy()
---> 82 self._last_mask = np.zeros(data.shape[1], dtype=np.bool)
     83 self.shape = self.data.shape
     84 self.supports_delta_masking = True

File E:\Anaconda\lib\site-packages\numpy\__init__.py:305, in __getattr__(attr)
    300     warnings.warn(
    301         f"In the future `np.{attr}` will be defined as the "
    302         "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    304 if attr in __former_attrs__:
--> 305     raise AttributeError(__former_attrs__[attr])
    307 # Importing Tester requires importing all of UnitTest which is not a
    308 # cheap import Since it is mainly used in test suits, we lazy import it
    309 # here to save on the order of 10 ms of import time for most users
    310 #
    311 # The previous way Tester was imported also had a side effect of adding
    312 # the full `numpy.testing` namespace
    313 if attr == 'testing':

AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

**Comment by znacer** [comment]
You should update your shap version to a version > [0.42](https://github.com/shap/shap/releases/tag/v0.42.0) (`pip install --upgrade shap` or `pip install "shap>=0.42"`). This deprecation has been fixed.

**Comment by connortann** [comment]
Closing as this issue was fixed in `0.42.0`
